MTA-6811, 6812, 6813, 6814, 6815, 6816 CQA April Developer Lightspeed work (#357)
migtools/mta-documentation: docs/topics/developer-lightspeed/proc_running-agent-analysis.adoc

= Generating code resolutions in the agent mode

[role="_abstract"]

> In the agent mode, the {mta-dl-plugin} planning agent creates the context for an issue and picks a sub-agent that is most suited to resolve the issue. The sub-agent runs an automated scan to describe how to resolve the issue and generates files with the updated resolutions in one stream. 

Reviewer: The current abstract is more of a feature-centric background to the user story, rather than the purpose/aim of the user story, which is not great for a procedural module. The 'meat' of the user story actually seems to be in the 2nd paragraph.

I suggest to rewrite and reshuffle the first 3 paragraphs, with the first one starting with the goal of the procedure (perhaps something like "To obtain and review code from Red Hat Developer Lightspeed for MTA, use..."), and only then proceed with describing the workings of the toolchain. This is a must for DITA and content usability especially visibility for SEO/AEO.

I've seen other similar abstracts, fix if it occurs elsewhere too.

You can review the updated files and approve or reject the changes to the code. The agent runs another automated analysis to detect new issues in the code that may have occurred because of the accepted changes or diagnostic issues that your tool may generate following an earlier analysis. If you allow the process to continue, {mta-dl-plugin} runs the stream again and generates a new file with the latest updates.

When using the agent mode, you can reject the changes or discontinue the stream but you cannot edit the updated files during the stream.

.Prerequisites

* You opened a Java project in your Visual Studio Code workspace.
* You configured an analysis profile on the *{ProductShortName} Analysis View* page.

.Procedure

. Verify that agent mode is enabled in one of the following ways:
+

> .. Type `Ctrl + Shift + P` in Visual Studio Code search (Linux/Windows system) and `Cmd + Shift + P` for Mac to go to the command palette. 

Reviewer: ```suggestion
.. Type `Ctrl+Shift +P` in Visual Studio Code search on Linux or Windows systems and `Cmd+Shift+P` for Mac to go to the command palette. 
```

Avoid using slashes in general text. Use 'and' or 'or' instead. See https://www.ibm.com/docs/en/ibm-style?topic=punctuation-slashes#general-text. 
Do not insert a space before or after the plus sign in key combinations. See https://www.ibm.com/docs/en/ibm-style?topic=elements-keyboard-keys#key-combinations.

Fit it if it occurs elsewhere too.

= Generating code resolutions in the agent mode

[role="_abstract"]

> As a Migrator, you can use the agent mode to run an automated analysis that streams the code resolutions that the agent can apply to your source code. When using the agent mode, you can reject the changes or stop the stream but you cannot edit the updated files during the stream.

Reviewer 2: ```suggestion
As a Migrator, to run an automated analysis that streams code resolutions, use the agent mode.

When using the agent mode, you can reject the changes or stop the stream. However,you cannot edit the updated files during the stream.
```

= Generating code resolutions in the agent mode

[role="_abstract"]
As a Migrator, to run an automated analysis that streams the code resolutions, use the agent mode. 

When using the agent mode, you can reject the changes or stop the stream. However, you cannot edit the updated files during the stream.

> The agent detects new issues in the code caused by earlier changes or diagnostic issues that your tool may generate. If you allow the process to continue, {mta-dl-plugin} runs the stream again and generates a new file with the latest updates.

Reviewer 2: IBM SG
```suggestion
The agent detects new issues in the code caused by earlier changes or diagnostic issues that your tool might generate. If you allow the process to continue, {mta-dl-plugin} runs the stream again and generates a new file with the latest updates.
```
